graph/src/data/schema.rs

```diff
@@ -606,6 +815,193 @@ impl Schema {
             })
     }
 
+    fn validate_fulltext_directives(&self) -> Vec<SchemaValidationError> {
+        self.subgraph_schema_object_type()
+            .map_or(vec![], |subgraph_schema_type| {
+                subgraph_schema_type
+                    .directives
+                    .iter()
+                    .filter(|directives| directives.name.eq("fulltext"))
+                    .fold(vec![], |mut errors, fulltext| {
+                        self.validate_fulltext_directive_name(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_language(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_algorithm(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_includes(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        errors
+                    })
+            })
+    }
+
+    fn validate_fulltext_directive_name(&self, fulltext: &Directive) -> Vec<SchemaValidationError> {
+        let name = match fulltext.argument("name") {
+            Some(Value::String(name)) => name,
+            _ => return vec![SchemaValidationError::FulltextNameUndefined],
+        };
+
+        let local_types: Vec<&ObjectType> = self
+            .document
+            .get_object_type_definitions()
+            .into_iter()
+            .collect();
+
+        // Validate that the fulltext field doesn't collide with any entity fields
```

Comment:
Thinking about this again: I think we should allow fulltext fields to have the same name as entity fields after all.

From a subgraph developer's perspective, there's no possibility of collisions because the fulltext search field is global (on the `Query` type) and entity fields are local (on the entity type). So we should be allowing overlaps and handle this internally.

My suggestion would be to map fulltext directives named `foo` to fields named `_fulltext_foo`. This has several benefits:

1. No collisions with entity fields possible (assuming we disallow fields with a leading `_`, which we will).
2. It's clear which fields come from which `_Schema_` directive (right now, there is only `fulltext`, but who knows what else we'll add).

Reply:
I'm not clear on your suggestion here, where would the fields be named like `_fulltext_foo`?
In the places I'm thinking of the fulltext fields are clearly marked, such as fulltext [Columns.](https://github.com/graphprotocol/graph-node/blob/ford/text/store/postgres/src/relational.rs#L734)

Reply:
I was thinking in the database schema that is generated. Say we have this entity:
```
type Foo @entity {
  bar: BigInt!
}
```
And now we add a fulltext directive called `"bar"`. Would this try to generate the `bar` column twice? Or would it generate `bar` and `_fulltext_bar` or something like that?